templates/web-application/package.json

```diff
@@ -0,0 +1,54 @@
+{
+  "name": "{{PACKAGE-NAME}}",
+  "version": "0.0.0",
+  "description": "",
+  "type": "module",
+  "scripts": {
+    "dev": "webpack serve --config webpack.dev.js",
+    "build": "webpack --config webpack.build.js",
+    "test": "jest",
+    "serve-build": "pnpm http-server dist -p 8080 -P http://localhost:8080?",
+    "reset": "pnpm dlx rimraf node_modules dist",
+    "lint": "eslint .",
+    "analyze": "webpack --config webpack.build.js --analyze",
+    "postinstall": "pnpm dlx rimraf node_modules/.cache"
+  },
+  "dependencies": {
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-router-dom": "6.11.0"
+  },
+  "devDependencies": {
+    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.10",
```

Comment:
I thought a little bit about this and one solution would be a tool that automatically set a strict version at generation but stay within the MAJOR version range, meaning it only updates MINOR and PATCH.

Any thoughts?

Reply:
That could be nice.

We can add in the CLI an end script that will do a `pnpm update` (it updates the versions in `packages.json` once completed) and then apply a  text replacement on the `package.json` that will remove the `^` or `~` in dependencies version to lock them.

What do you think of this as a solution?

Reply:
That's an interesting solution but I am concerned that it could drastically increase the generation time. How long would that take to execute a `pnpm update` ?